resources/ini.ebnf

```diff
@@ -0,0 +1,21 @@
+ini = body section*.
+
+section = header (eol body)?.
+header = "[" wsp name wsp "]" wsp comment?.
+name = #"[\w ]+(?<! )"
+
+body = (kv? wsp comment? eol)* (kv wsp comment?)?
+
+kv = key wsp "=" wsp val.
+key = #"[\w ]*(?<! )"
+val = #"[\w ]*(?<! )"
```

Comment:
My regex skills aren’t quite up to snuff to fully grok these. I tried pasting the pattern into [RegExr](https://regexr.com/) and read its explanation, but I didn’t get it. My intuition says this’ll match an empty string? And playing around with `re-seq` in a `bb` repl, that does seem to be the case? If that’s right, how can an empty string be a valid key?

(If I’m not an idiot, yet this is correct and there’s something subtle/specific going on here, consider adding a comment to explain.)

Reply:
Your read is correct and I have added a test to clarify that this is (mildly) intentional behavior. Mostly I'm trying to get as few failures as possible in the _parse_ stage, because that makes for maximally helpful partial parses. I agree that an empty key is... kind of nonsense though.